app/views/users/show.html.erb

```diff
@@ -3,7 +3,7 @@
     <%= render "users/user", user: @user %>
   </div>
 </div>
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
Might want to use a different method for this, since this page gets rendered from the user#show action, there is never a case where this is hidden and the whole page is not authorized